docs/source/release_notes/release-notes.rst

```diff
@@ -43,6 +43,11 @@ develop
          - Change
 
     *    - |fixed|
+         - ``PaxosTimestampBoundStore`` now throws ``NotCurrentLeaderException``, invalidating the timestamp store, if a bound update fails because another timestamp service on the same node proposed a smaller bound for the same sequence number.
```

Comment:
Okay. I'm going to open against 0.42.x first for expediency as you say, which cleans up the release notes as required.